WordPress/src/main/java/org/wordpress/android/sharedlogin/JetpackPublicData.kt

```diff
@@ -0,0 +1,38 @@
+package org.wordpress.android.sharedlogin
+
+import org.wordpress.android.BuildConfig
+import org.wordpress.android.sharedlogin.JetpackPublicData.PackageName.Jalapeno
+import org.wordpress.android.sharedlogin.JetpackPublicData.PackageName.Vanilla
+import org.wordpress.android.sharedlogin.JetpackPublicData.PackageName.Wasabi
+import org.wordpress.android.sharedlogin.JetpackPublicData.PublicKeyHash.Debug
+import org.wordpress.android.sharedlogin.JetpackPublicData.PublicKeyHash.Release
+import javax.inject.Inject
+
+class JetpackPublicData @Inject constructor() {
```

Comment:
Not sure what's your opinion on this, but this could have been also implemented with flavour-specific `buildConfigField`'s in `WordPress-Android/build.gradle`, removing the need of wrapping logic to retrieve the correct value.

Just wanted to bring it into discussion here if it wasn't considered yet :)

Reply:
Thanks for the review, @ovitrif !

>Another thing that made me hesitate was the TODO under test case 4 - To test 1, 2 and 3 with release versions.  

Makes sense! I was struggling a bit to understand how to test case 4. I've managed to test it by uploading the bundles to Play Store (thanks for the help @AliSoftware ) and it's working as expected for release version too.

>I added a comment about using string buildConfigField's to get the required values that we're currently getting with logic that needs unit testing.
>Let me know what's your take on that.

Good point! @AliSoftware had a similar idea, but to actually extract the values to strings (`src/{debug,release}/res/`) with `translatable="false"`. What I told him is that by having the values on a class I'm able to [unit test them](https://github.com/wordpress-mobile/WordPress-Android/pull/17059/files#diff-c1bd78ba29be521e14f191663aac258a6584e8a852d385d6b7c7b3270bd39de9). IMHO this is good because if someone by accident changes the value in the class, the test will fail. But since both of you had kind of similar ideas, I don't mind changing it if you believe it works best 🙂

Reply:
I think the `BuildConfigField` idea is actually better than my idea of using `src/{debug,release}/res/values` 👍  as those constants feels more in the territory of built-time constants — provided by Gradle, and driven by which flavor we use — than runtime resources and using `R.string.` values (which don't feel like the best tool for a non-UI-related constant after all) 🙃